### PR TITLE
Replace unmaintained Hurley gem with a combination of Faraday and HTTP

### PR DESCRIPTION
**Pull request by cbeer**
Deployed to -stage and seems to work fine.

### CODE REVIEW

File: app/models/concerns/djatoka_adapter.rb [hunk 1]
```diff
@@ -2,14 +2,8 @@
 # Djatoka-backed implementation of StacksImage delivery
 module DjatokaAdapter
   def response
```
Comment (jcoyne) [inline app/models/concerns/djatoka_adapter.rb]: This is never called with a block?

File: app/models/concerns/djatoka_adapter.rb [hunk 2]
```diff
@@ -2,14 +2,8 @@
...
-    return to_enum(:response) unless block_given?
-
     benchmark "Fetch #{url}" do
-      client.get(url) do |req|
-        req.on_body do |_, chunk|
-          yield chunk
-        end
-      end
+      HTTP.get(url).body
```
Comment (jkeck) [inline app/models/concerns/djatoka_adapter.rb]: Why `HTTP` here instead of `Faraday`?

Comment (cbeer) [inline app/models/concerns/djatoka_adapter.rb]: `HTTP::Reponse#body` streams the response back as an IO-like object. Faraday doesn't (yet) support content streaming.

File: app/models/concerns/djatoka_adapter.rb (hunk 1 above)
Comment (jcoyne) [inline app/models/concerns/djatoka_adapter.rb]: Add documentation that this method is returning an IO.

File: app/models/concerns/djatoka_adapter.rb (hunk 2 above)
Comment (jkeck) [inline app/models/concerns/djatoka_adapter.rb]: Thanks for the clarification.

File: app/models/concerns/djatoka_adapter.rb (hunk 1 above)
Comment (cbeer) [inline app/models/concerns/djatoka_adapter.rb]: Done.

**Pull request merged**